pkg/ddl/partition.go

```diff
@@ -3127,16 +3119,22 @@ func getReorgPartitionInfo(t *meta.Mutator, job *model.Job) (*model.TableInfo, [
 //	Everything now looks as it should, no memory of old partitions/indexes,
 //	and no more double writing, since the previous state is only reading the new partitions/indexes.
 func (w *worker) onReorganizePartition(jobCtx *jobContext, job *model.Job) (ver int64, _ error) {
+	args, err := model.GetTablePartitionArgs(job)
+	if err != nil {
+		job.State = model.JobStateCancelled
+		return ver, errors.Trace(err)
+	}
+	jobCtx.jobArgs = args
 	// Handle the rolling back job
 	if job.IsRollingback() {
-		ver, err := w.rollbackLikeDropPartition(jobCtx, job)
+		ver, err := w.rollbackLikeDropPartition(jobCtx, job, args)
```

Comment:
No need to pass `args`?

Reply:
indeed https://github.com/pingcap/tidb/pull/56656/commits/bb592f81ef59dddb9a518b1253297b503631f6ff